src/components/Company/Offers/Manage/CompanyOffersTitle.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import { Chip } from "@material-ui/core";
+
+export const statusChips = {
+    hidden: <Chip label="Hidden" size="small" style={{ backgroundColor: "#90A4AE" }} />,
+    blocked: <Chip label="Blocked" size="small" style={{ backgroundColor: "#DC4338" }} />,
+    archived: <Chip label="Archived" size="small" style={{ backgroundColor: "#56A8D6" }} />,
+};
+
+const OfferTitle = ({ title, offersVisibility, setOfferVisibility, offerId }) => {
+    const [chips, setChips] = useState([]);
+    const isHidden = offersVisibility[offerId]?.isHidden;
+    const isBlocked = offersVisibility[offerId]?.isDisabled;
+    const isArchived = offersVisibility[offerId]?.isArchived;
+
+    useEffect(() => {
+        const tempChips = [];
+        if (isHidden)
+            tempChips.push(statusChips.hidden);
+        if (isBlocked)
+            tempChips.push(statusChips.blocked);
+        if (isArchived)
+            tempChips.push(statusChips.archived);
+        setChips(tempChips);
+    }, [isArchived, isBlocked, isHidden, offerId, offersVisibility, setOfferVisibility]);
+
+    return (
+        <>
+            {title}
+            <div style={{ position: "absolute" }}>
+                {chips.reduce((res, chip, index, chips) => {
+                    res.push(chip);
+                    if (index !== chips.length - 1)
+                        res.push(<span>&nbsp;</span>);
+                    return res;
+                }, [])}
```

Comment:
Normally this pattern (taking an array of objects and creating an array of JSX elements, which is parsed directly by JSX engines) is done through the use of the `map` function: you map each data element to a UI element. I'll give an example:

```js
function Component({ ... }) {
    ...
    const data = [
        { label: "", url: "" },
        { label: "", url: "" },
    ];
    ...

    return (
        {
            data.map(({ label, url }) => (
                <a href={url}>{label}</a>
            ))
        }
    );
}
```

If I understood your approach, you made use of the `reduce` function to introduce the "non-breaking space" character but I fail to see its use here in favor of, say, styling the elements.

I'm not saying this is wrong, only that I found it strange. I'm leaving this as a change request but please do argue about this.